### ISSUE TITLE
Specified username login failed

### ISSUE THREAD

**Issue by leotu** [vert-x3/vertx-redis-client#348]
### Questions

Log in to Redis instead of using the default user, use the specified username connected failed

For example:
[app.redislabs.com](https://app.redislabs.com/)
Would create specified username & password

### Version

```xml
  <artifactId>vertx-redis-client</artifactId>
  <version>4.3.2</version>
```

### Context

[Redis.createClient](https://github.com/vert-x3/vertx-redis-client/blob/223b351b513ee671924bed1345040fad6945cd7a/src/main/java/io/vertx/redis/client/Redis.java#L46)

```text
redis://[username:password@][host][:port][/database]
```

Maybe some problem at:

Redis ACL (https://redis.io/docs/manual/security/acl/)

```txt
AUTH <username> <password>
```

### Steps to reproduce

1. Test Code

```java
class TestRedisClient {
    private static final Logger log = LoggerFactory.getLogger(TestRedisClient.class);

    @Test
    void test() {
        log.debug("Begin...");
        CountDownLatch latch = new CountDownLatch(1);
        try {
            Vertx vertx = Vertx.vertx();

            String username = URLEncoder.encode("xxx", StandardCharsets.UTF_8);
            String password = URLEncoder.encode("zzz", StandardCharsets.UTF_8);

            String connectionString = "redis://" + username + ":" + password + "@redis-10770.c60.us-west-1-2.ec2.cloud.redislabs.com:10770/0";
            
           Redis.createClient(vertx, connectionString)
                    .connect()
                    .onSuccess(conn -> {
                        log.debug("onSuccess, conn: {}", conn);
                        latch.countDown();
                    })
                    .onFailure(err -> {
                        log.error("onFailure", err);
                        latch.countDown();
                    });

            log.debug("Before await");
            latch.await();
            log.debug("After await");
        } catch (Exception e) {
            log.error(e.toString());
        } finally {
            log.debug("End.");
        }
    }
}
```

 2. Console error log

```
23:03:43.231 DEBUG [main] [com.xgosh.test.TestRedisClient:20,test()] Begin...
23:03:43.484 DEBUG [main] [com.xgosh.test.TestRedisClient:39,test()] Before await
23:03:44.245 ERROR [vert.x-eventloop-thread-0] [com.xgosh.test.TestRedisClient:35,lambda$1()] onFailure: WRONGPASS invalid username-password pair

23:03:44.247 DEBUG [main] [com.xgosh.test.TestRedisClient:41,test()] After await
23:03:44.247 DEBUG [main] [com.xgosh.test.TestRedisClient:45,test()] End.
```

### Extra Info
- https://app.redislabs.com/

<img width="1434" alt="截圖 2022-09-16 22 30 36" src="https://user-images.githubusercontent.com/1053138/190666930-bc839d77-7575-4c3c-bf35-eef29e969b30.png">

<img width="1285" alt="截圖 2022-09-16 22 31 10" src="https://user-images.githubusercontent.com/1053138/190666953-76282fba-0071-4d5a-b967-45d51bb23465.png">

**Comment by pmlopes** [comment]
@leotu are you encoding the username / password special chars in URL format? If that still doesn't work can you privately provide a test account for me to test and debug the issue?

**Comment by leotu** [comment]
@pmlopes 

I create new account for your

```java
String username = URLEncoder.encode("demo", StandardCharsets.UTF_8);
String password = URLEncoder.encode("demo123~X", StandardCharsets.UTF_8);

String connectionString = "redis://" + username + ":" + password + "@redis-14177.c11.us-east-1-2.ec2.cloud.redislabs.com:14177/0";

// "redis://demo:demo123~X@redis-14177.c11.us-east-1-2.ec2.cloud.redislabs.com:14177/0"
```

**Comment by pmlopes** [comment]
@leotu I'm running the test with the latest git code and I can connect and execute commands such as `INFO`. I believe the issue was fixed on ac116b7bb7999593e787804a76cd55fb2291c3c6

**Comment by leotu** [comment]
Great !

I face this problem when that I run Quarkus version "2.12.2.Final", but it includes "vertx-redis-client" version "4.3.2".
([redis-reference](https://quarkus.io/guides/redis-reference))
